pkg/ingester/ingester.go

```diff
@@ -907,6 +912,8 @@ func (i *Ingester) pushSamplesToAppender(userID string, timeseries []mimirpb.Pre
 
 	// Return true if handled as soft error, and we can ingest more series.
 	handleAppendError := func(err error, timestamp int64, labels []mimirpb.LabelAdapter) bool {
+		stats.failedSamplesCount++
```

Comment:
Note to reviewers: I moved it here because it was always incremented right before calling `handleAppendError()`.